client/app/components/ParameterValueInput.jsx

```diff
@@ -181,15 +244,21 @@ export default function init(ngModule) {
         enum-options="$ctrl.param.enumOptions"
         query-id="$ctrl.param.queryId"
         on-select="$ctrl.setValue"
+        apply-button="$ctrl.applyButton"
       ></parameter-value-input-impl>
     `,
     bindings: {
       param: '<',
+      applyButton: '=?',
+      onChange: '=',
     },
     controller($scope) {
       this.setValue = (value) => {
         this.param.setValue(value);
-        $scope.$applyAsync();
+        $scope.$apply();
```

Comment:
@arikfr changing `$applyAsync` to `$apply` here did the trick for the parameters not working on dashboards. What happens is that dashboards depends on Url Parameters to be updated (and it's a `watch` that does it) and the refresh was being triggered before.

Reply:
Luckily soon we will get rid of all these crap 😎